### PR TITLE
Fix Entity directory does not exist

### LINKED ISSUE #299: No directory for Entity
While making PHPDocs for Maker Bundle, I found, that if I'm creating, for example, new Command and there is no `Command` directory in application, Maker creates it, but in a case of new Entity - it doesn't.

`alexeysamara: ~/projects/symfony-maker-bundle/dev (psr-fix**) $ c make:entity`

```
2018-10-21T13:54:30+00:00 [error] Error thrown while running command "make:entity". Message: "The "/Users/alexeysamara/projects/symfony-maker-bundle/dev/src/Entity/" directory does not exist."

In Finder.php line 547:

  The "/Users/alexeysamara/projects/symfony-maker-bundle/dev/src/Entity/" directory does not exist.
```

I'm not sure - this is bug or feature )

### PR DESCRIPTION
**Pull request by yceruto**
Closes #299 according to https://github.com/symfony/maker-bundle/pull/185#issuecomment-389009674

Also alternative for https://github.com/symfony/maker-bundle/pull/185, https://github.com/symfony/maker-bundle/pull/265 and https://github.com/symfony/maker-bundle/pull/310

### CODE REVIEW

**Comment by weaverryan** [comment]
OMG - finally! So happy to see this PR! And, it was even simpler than I expected!

We do have test failures, unfortunately. @yceruto could you take a look at them? They could be false failures. But it might also be an indication that using the new way may cause an unintended exception to be thrown in some cases.

Thanks!

**Comment by weaverryan** [comment]
Also, see #310 - I think the other PR may have "hit" the same issue that's causing the tests to fail here 🤔

**Comment by yceruto** [comment]
Yep, (in the same life time of the script) the first time that `$this->doctrineHelper->getEntitiesForAutocomplete()` is called with no entities some protected Doctrine properties are cached (e.g. `AnnotationDriver::$classNames`) and there is no way to invalidate it without force it and discover the metadatas of the new generated Entity.

Even if the last commit resolves part of the problem I guess there are still other cached properties related to the mapping type that need to be invalidated for the same reason.

So I think we should go back to the `Finder` version and use `AnnotationDriver::getPaths()` to discover the mapping Entity directory, wdyt?

**Comment by yceruto** [comment]
Also we could re-think the way this Annotation Mapping requirement is validated, because now it's checked after the entity was already generated.

**Comment by yceruto** [comment]
Well, I've fixed an invalid mapping from the fixture of the two remaining fails and all green again.

File: src/Doctrine/DoctrineHelper.php [hunk 1]
```diff
@@ -145,6 +146,19 @@ public function getMetadata(string $classOrNamespace = null, bool $disconnected
                 foreach ($loaded as $m) {
                     $cmf->setMetadataFor($m->getName(), $m);
                 }
+
+                // Invalidating the cached AnnotationDriver::$classNames to find new Entity classes
+                $metadataDriver = $em->getConfiguration()->getMetadataDriverImpl();
+                if ($metadataDriver instanceof MappingDriverChain) {
+                    foreach ($metadataDriver->getDrivers() as $driver) {
+                        if ($driver instanceof AnnotationDriver) {
```
Comment (weaverryan) [inline src/Doctrine/DoctrineHelper.php]: Wow! Yea, this issue is a bummer - but nice job at least finding a workaround.

Does this only affect the AnnotationDriver?

Comment (yceruto) [reply src/Doctrine/DoctrineHelper.php]: And also for `StaticPHPDriver` but we only support annotation driver, right?

**Comment by weaverryan** [comment]
Thank you very much for figuring this out @yceruto!

**Pull request merged**